### PR TITLE
 feat: Add maximum nesting depth calculator for Python functions

### LINKED ISSUE #141: [FEATURE] Implement nesting depth calculation
### Background
  We want to add **nesting depth** as a new complexity metric to help developers identify deeply nested code that
   is hard to understand.

  Nesting depth measures how deeply code structures are nested (if statements, loops, try blocks, etc.). For
  example:

  ```python
  def example():
      if condition:           # depth 1
          for item in items:  # depth 2
              while x > 0:    # depth 3
                  pass
```
  This function has a maximum nesting depth of 3.

  ### Task
  Implement a function that calculates the maximum nesting depth for a Python function by traversing its AST.

  ### Implementation
Create new file: internal/analyzer/nesting_depth.go

### Additional context
  1. AST Traversal: Look at internal/parser/visitor.go for examples of traversing AST nodes
  2. Algorithm: Use recursion - increment depth when entering a nesting node, decrement when leaving
  3. Track maximum: Keep track of the maximum depth encountered during traversal
  4. Getting function nodes: Check internal/analyzer/complexity_analyzer.go to see how function nodes are
  extracted

### Resources
  - ../internal/parser/visitor.go
  - ../internal/parser/ast.go
  - ../internal/analyzer/complexity_analyzer.go

  ### Getting started

  1. Fork the repository
  2. Create a new branch: git checkout -b feat/nesting-depth
  3. Submit a PR

### PR DESCRIPTION
**Pull request by GergesHany**
## Description
Implements a function to calculate the maximum nesting depth for Python functions by traversing their Abstract Syntax Tree (AST).

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Related Issues
Closes #141 

## Changes Made

- **New file**: `internal/analyzer/nesting_depth.go`
  - `CalculateMaxNestingDepth()`: Main function that computes max nesting depth for a single Python function
  - `traverseForNesting()`: Recursive AST traversal helper
  - `isNestingNode()`: Determines which node types increase nesting depth
  - `isComprehensionNode()`: Helper to identify comprehension nodes
  - `NestingDepthResult`: Struct holding nesting depth metrics and location info

- **New file**: `internal/analyzer/nesting_depth_test.go`
  - Comprehensive test coverage with 10+ test cases
  - Tests for various nesting scenarios: if/for/while/try/with statements, comprehensions, lambdas
  - Edge cases: nil nodes, simple functions, complex nesting (depth 4+)

## Testing
- [x] Unit tests pass locally
- [ ] Integration tests pass locally
- [ ] Manual testing completed

## Checklist
- [x] My code follows the project's style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Screenshots (if applicable)

image

### CODE REVIEW

**Comment by DaisukeYoda** [comment]
@GergesHany 

Great work! I have one question about the nesting node definition (commented inline).

File: internal/analyzer/nesting_depth.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+// isNestingNode determines if a node type increases nesting depth
+func isNestingNode(node *parser.Node) bool {
+	if node == nil {
+		return false
+	}
+
+	switch node.Type {
+	case parser.NodeIf, parser.NodeFor, parser.NodeAsyncFor, parser.NodeWhile, parser.NodeWith, parser.NodeAsyncWith, parser.NodeTry, parser.NodeExceptHandler, parser.NodeMatch, parser.NodeMatchCase, parser.NodeElifClause,
```
[the remark this answers is not in the document]
Comment (DaisukeYoda) [reply internal/analyzer/nesting_depth.go:114]: Thanks!

**Review by DaisukeYoda — APPROVED** [review]
